projects/RabbitMQ.Client/client/impl/RabbitMQActivitySource.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading;
+using RabbitMQ.Client.Events;
+using RabbitMQ.Client.Framing.Impl;
+
+namespace RabbitMQ.Client
+{
+    internal static class RabbitMQActivitySource
+    {
+        internal const string MessageId = "messaging.message.id";
+        internal const string MessageConversationId = "messaging.message.conversation_id";
+        internal const string MessagingOperation = "messaging.operation";
+        internal const string MessagingSystem = "messaging.system";
+        internal const string MessagingDestination = "messaging.destination.name";
+        internal const string MessagingDestinationRoutingKey = "messaging.rabbitmq.destination.routing_key";
+        internal const string MessagingPayloadSize = "messaging.message.payload_size_bytes";
+        internal const string ProtocolName = "net.protocol.name";
+        internal const string ProtocolVersion = "net.protocol.version";
+
+
+        internal static readonly ActivitySource s_source = new ActivitySource("RabbitMQ.Client", typeof(RabbitMQActivitySource).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion);
+
+        internal static Activity Send(string routingKey, string exchange, int bodySize)
+        {
+            if (s_source.HasListeners())
+            {
+                Activity activity = StartRabbitMQActivity($"{routingKey} publish", ActivityKind.Producer, Activity.Current?.Context ?? default);
```

Comment:
This activity name is likely to be inappropriate in a lot of cases if there is high cardinality in the routing key. For example, the routing key may have a segment for a user id or even a unique value like request id that would result in an un-useful number of different activity names.

There needs to be a way to conditionally/optionally override the Activity naming so that a routing key like `response.1435266` could be labeled as `response.{requestId}` or some other static string through provided logic.

Reply:
Added a configuration setting to determine if the routing key should be part of the span name or not.